File: library/src/main/java/com/chaek/android/adapter/OnRecyclerItemClickListener.java
package com.chaek.android.adapter;

/**
 * RecyclerView Adapter item onClick Listener
 *
 * @author Chaek
 */
public interface OnRecyclerItemClickListener {
    /**
     * onClick callback
     *
     * @param t     click item data
     * @param index click item position
     */
    void onClick(Object t, int index);
}

File: library/src/main/java/com/chaek/android/adapter/OnRecyclerItemClickListener.kt
package com.chaek.android.adapter

/**
 * RecyclerView Adapter item onClick Listener
 *
 * @author Chaek
 */
interface OnRecyclerItemClickListener<T> {
    /**
     * onClick callback
     *
     * @param t     click item data
     * @param index click item position
     */
    fun onClick(t: T, index: Int)

}
